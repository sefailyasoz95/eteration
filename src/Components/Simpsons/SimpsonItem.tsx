import React, { useEffect, useState } from "react";
import { View, Text, StyleSheet, Image, TouchableOpacity, GestureResponderEvent, Alert } from "react-native";
import { imageUrlHelper } from "../../Utils/Helpers/imageUrlHelper";
import { deleteSimpson } from "../../Utils/Services/service";
type Props = {
	item: any;
	onPress: (event: GestureResponderEvent) => void;
	onDelete: (event: GestureResponderEvent) => void;
};

const SimpsonItem: React.FC<Props> = ({ item, onPress, onDelete }) => {
	const [isValidURL, setIsValidURL] = useState(false);
	useEffect(() => {
		setIsValidURL(imageUrlHelper(item.avatar));
	}, []);

	return (
		<TouchableOpacity style={styles.itemContainer} onPress={() => onPress(item)}>
			<View style={styles.imageAndName}>
				{isValidURL ? (
					<Image source={{ uri: item.avatar }} width={50} height={50} style={styles.image} />
				) : (
					<Text style={styles.noImage}>â›”</Text>
				)}
				<Text style={styles.name}>{item.name}</Text>
			</View>
			<TouchableOpacity onPress={() => onDelete(item)}>
				<Text style={styles.deleteButton}>{`x`} </Text>
			</TouchableOpacity>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	itemContainer: {
		padding: 10,
		marginVertical: 5,
		flexDirection: "row",
		alignItems: "center",
		justifyContent: "space-between",
		marginHorizontal: 5,
		backgroundColor: "#fff",
		elevation: 5,
		shadowColor: "black",
		shadowOffset: {
			width: 0,
			height: 1,
		},
		shadowOpacity: 0.3,
		shadowRadius: 3,
		borderRadius: 10,
	},
	image: {
		height: 50,
		width: 50,
		resizeMode: "contain",
	},
	noImage: {
		padding: 10,
	},
	name: {
		letterSpacing: 1,
		marginHorizontal: 10,
	},
	imageAndName: {
		flexDirection: "row",
		alignItems: "center",
		justifyContent: "flex-start",
	},
	deleteButton: {
		fontSize: 25,
		fontWeight: "bold",
		color: "#ff2222",
		padding: 10,
	},
});

export default SimpsonItem;
