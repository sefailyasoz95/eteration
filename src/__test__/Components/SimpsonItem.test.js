import React from "react";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { render } from "@testing-library/react-native";
import SimpsonItem from "../../Components/Simpsons/SimpsonItem";
import { create } from "react-test-renderer";

configure({ adapter: new Adapter() });

const simpsons = [
	{ id: 1, name: "Merge Simpson", about: "something about Merge", job: "Merge's job", avatar: "https://image.com" },
	{ id: 2, name: "Homer Simpson", about: "something about Homer", job: "Homer's job", avatar: "https://image.com" },
];
const simpson = {
	id: 1,
	name: "Merge Simpson",
	about: "something about Merge",
	job: "Merge's job",
	avatar: "https://image.com",
};

const navigation = {
	navigate: jest.fn(),
};

describe("Simpson Item Component", () => {
	it("should render component correctly", () => {
		const component = shallow(<SimpsonItem item={simpsons[0]} />);

		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});

	it("should render correctly first item of the simpsons array", () => {
		const wrapper = render(<SimpsonItem item={simpsons[0]} />);
		expect(wrapper.UNSAFE_getAllByType(SimpsonItem).length).toBe(simpsons.length - 1);
	});
	it("should render correct names", () => {
		const wrapper = shallow(<SimpsonItem item={simpson} />);

		wrapper.find({ testID: "simpsonName" }).forEach((node, index) => {
			expect(node.props().children).toBe(simpson.name);
		});
	});
	it("it should simulate onPress and passing object to parent object which is SimpsonsList ", () => {
		const component = create(<SimpsonItem item={simpsons[0]} onPress={navigation.navigate} />);
		const opacity = component.root.findByProps({ testID: "simpsonItem" }).props;
		opacity.onPress();

		expect(navigation.navigate).toBeCalledWith(simpson);
	});
});
