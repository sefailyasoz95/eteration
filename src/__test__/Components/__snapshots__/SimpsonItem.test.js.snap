// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simpson Item Component should render component correctly 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 5,
      "marginVertical": 5,
      "padding": 10,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
  testID="simpsonItem"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "padding": 10,
        }
      }
    >
      â›”
    </Text>
    <Text
      style={
        Object {
          "letterSpacing": 1,
          "marginHorizontal": 10,
        }
      }
      testID="simpsonName"
    >
      Merge Simpson
    </Text>
  </View>
  <ForwardRef
    onPress={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#ff2222",
          "fontSize": 25,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      x
       
    </Text>
  </ForwardRef>
</ForwardRef>
`;
